libraries/Microsoft.Bot.Builder.Dialogs/DialogContainer.cs

```diff
@@ -49,6 +50,31 @@ public virtual Dialog FindDialog(string dialogId)
             return this.Dialogs.Find(dialogId);
         }
 
+        /// <summary>
+        /// Called when an event has been raised, using `DialogContext.emitEvent()`, by either the current dialog or a dialog that the current dialog started.
+        /// </summary>
+        /// <param name="dc">The dialog context for the current turn of conversation.</param>
+        /// <param name="e">The event being raised.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>True if the event is handled by the current dialog and bubbling should stop.</returns>
+        public override async Task<bool> OnDialogEventAsync(DialogContext dc, DialogEvent e, CancellationToken cancellationToken)
+        {
+            var handled = await base.OnDialogEventAsync(dc, e, cancellationToken).ConfigureAwait(false);
+
+            // Trace unhandled "versionChanged" and "error" events.
+            if (!handled && (e.Name == DialogEvents.VersionChanged || e.Name == DialogEvents.Error))
+            {
+                var traceMessage = $"Unhandled dialog event: {e.Name}. Active Dialog: {dc.ActiveDialog.Id}";
+
+                dc.Dialogs.TelemetryClient.TrackTrace(traceMessage, Severity.Warning, null);
```

Comment:
Is the amount of information conveyed here actionable for a developer?

Note: This might not be a question for @garypretty, but rather the architects of the DialogEvent-ing system. @tomlm, @Stevenic?

Reply:
@garypretty, could you please check the changes requested?